zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs

```diff
@@ -340,61 +340,102 @@ mod tests {
     use eth_types::{ToScalar, U256};
     use halo2_proofs::{circuit::Value, halo2curves::bn256::Fr};
 
-    // <https://github.com/scroll-tech/go-ethereum/blob/develop/rollup/fees/rollup_fee_test.go>
-    const TEST_BASE_FEE: u64 = 15_000_000;
+    //refer to test in <https://github.com/scroll-tech/go-ethereum/blob/develop/rollup/fees/rollup_fee_test.go#L10>
+    const TEST_BASE_FEE_BEFORE_CURIE: u64 = 15_000_000;
     const TEST_FEE_OVERHEAD: u64 = 100;
     const TEST_FEE_SCALAR: u64 = 10;
     const TEST_TX_DATA_GAS_COST: u64 = 40; // 2 (zeros) * 4 + 2 (non-zeros) * 16
-    const TEST_TX_L1_FEE: u128 = 30;
+    const TEST_TX_L1_FEE_BEFORE_CURIE: u128 = 30;
+
+    const TEST_AFTER_CURIE: u64 = 1;
+    const TEST_BEFORE_CURIE: u64 = 0;
+
+    // refer to test in https://github.com/scroll-tech/go-ethereum/blob/develop/rollup/fees/rollup_fee_test.go#L22
+    const TEST_BASE_FEE_AFTER_CURIE: u64 = 1_500_000_000;
+
+    const L1_BLOB_BASEFEE: u64 = 150_000_000;
+    const COMMIT_SCALAR: u64 = 10;
+    const BLOB_SCALAR: u64 = 10;
+
+    const TEST_TX_RLP_SIGNED_LENGTH: u128 = 4;
+    const TEST_TX_L1_FEE_AFTER_CURIE: u128 = 21;
 
     #[test]
     fn test_tx_l1_fee_with_right_values() {
-        let witnesses = [
-            TEST_BASE_FEE.into(),
-            TEST_FEE_OVERHEAD.into(),
-            TEST_FEE_SCALAR.into(),
-            TEST_TX_DATA_GAS_COST.into(),
-            TEST_TX_L1_FEE,
-        ]
-        .map(U256::from);
-
-        try_test!(TxL1FeeGadgetTestContainer<Fr>, witnesses, true);
+        // test both before & after curie upgrade
+        for is_curie in [TEST_BEFORE_CURIE, TEST_AFTER_CURIE] {
+            let witnesses = [
```

Comment:
the array is a bit difficult to maintain.  I would suggest a Struct

Reply:
you mean the `witness`  array , right ?

Reply:
yes

Reply:
emm, the array type is specified by the common test util `test_math_gadget_container` which this tests utilizes.  even though can make structure here, but finally still need to convert to array type. 

/```
// A simple macro for less code & better readability
macro_rules! try_test {
    ($base_class:ty, $witnesses:expr, $expect_success:expr $(,)?) => {{
        test_math_gadget_container::<Fr, $base_class>($witnesses.to_vec(), $expect_success)
    }};
}
```

Reply:
i see.. then current code is also ok